src/prefect/_internal/pydantic/_compat.py

```diff
@@ -13,14 +13,28 @@
 IncEx: TypeAlias = "Union[Set[int], Set[str], Dict[int, Any], Dict[str, Any], None]"
 logger = get_logger("prefect._internal.pydantic")
 
+
 if HAS_PYDANTIC_V2:
-    from pydantic import TypeAdapter, parse_obj_as
+    if USE_PYDANTIC_V2:
+        from pydantic import Field as PydanticField
+        from pydantic import PrivateAttr as PydanticPrivateAttr
+        from pydantic import TypeAdapter
+    else:
+        from pydantic.v1 import Field as PydanticField
+        from pydantic.v1 import PrivateAttr as PydanticPrivateAttr
+    from pydantic import parse_obj_as
```

Comment:
sorry why is `parse_obj_as` not being imported from V1 here? Is it just because it's virtually unchanged but deprecated?